pkg/operator/k8sutil/deployment.go

```diff
@@ -73,6 +90,40 @@ func UpdateDeploymentAndWait(context *clusterd.Context, deployment *apps.Deploym
 		}
 		if d.Status.ObservedGeneration != original.Status.ObservedGeneration && d.Status.UpdatedReplicas > 0 && d.Status.ReadyReplicas > 0 {
 			logger.Infof("finished waiting for updated deployment %s", d.Name)
+
+			// Now we check if we can go to the next daemon
+			actionToCheck := "continue"
+			for _, daemon := range daemonsToCheckList {
+				if strings.Contains(deployment.Name, daemon) {
+					logger.Infof("checking if deployment %s update can continue", deployment.Name)
+					err := upgrade.OkToStopOrContinue(context, namespace, deployment.Name, daemon, actionToCheck, clusterInfo.CephVersion)
+					if err != nil {
+						return nil, fmt.Errorf("failed to check if deployment %s can be updated. %+v", deployment.Name, err)
```

Comment:
I just tested updating from 13.2.5 to 14.2.0 and saw a failure:
```
failed to update osd deployment 0. failed to check if deployment rook-ceph-osd-0 can be updated. failed to check if rook-ceph-osd-0 was ok to continue
```

After this the operator restarted the orchestration and was able to complete successfully on the next attempt. Do we need to add a retry here until it's ok to continue instead of returning the error immediately?